package/src/components/AttachmentPicker/AttachmentPicker.tsx

```diff
@@ -245,14 +246,8 @@ export const AttachmentPicker = React.forwardRef(
       numberOfUploads: selectedFiles.length + selectedImages.length,
       // `id` is available for Expo MediaLibrary while Cameraroll doesn't share id therefore we use `uri`
       selected:
-        selectedImages.some((image) =>
-          image.id
-            ? image.id === asset.id
-            : image.uri === asset.uri || image.originalUri === asset.uri,
-        ) ||
-        selectedFiles.some((file) =>
-          file.id ? file.id === asset.id : file.uri === asset.uri || file.originalUri === asset.uri,
-        ),
+        selectedImages.some((image) => image.uri === asset.uri) ||
```

Comment:
Why are we doing this change exactly ?

Reply:
We don't rely on asset ID or originalUri(which is gone now), which is irrelevant for the new changes and are not supported by the RNFile type anyways. This can be pretty much checked from the file URI only and we can avoid the complex check here.

Reply:
We generate an id when we upload an image or file anyways so no point of getting an id from the native API here IMO.

Reply:
I'm talking about the `file.id` specifically. Let's please not introduce such changes (that we haven't tested properly) in larger PRs like this that involve quite a bit of refactoring.

Reply:
Since we are not passing ID anymore and RNFile doesn't have a property ID, you can't check for it ideally and talking about uri, this works as far as I have tested.